plugins/modules/webhook.py

```diff
@@ -0,0 +1,161 @@
+      - Required when creating a new webhook
+    type: str
+  webhook_template_id:
+    description:
+      - ID of the webhook template
+    type: str
+  enabled:
+    description:
+      - Enable or disable the webhook
+    type: bool
+  verify_ssl:
+    description:
+      - Verify SSL certs for the webhook
+    type: bool
+  ssl_ca_certs:
+    description:
+      - X509 Certification Authorities concatenated in PEM format
+    type: str
+  webhook_username:
+    description:
+      - Username for the webhook, if required
+    type: str
+  webhook_password:
+    description:
+      - Password for the webhook, if required
+    type: str
+  http_headers:
+    description:
+      - HTTP headers for the webhook
+    type: str
+  proxy_authorization:
+    description:
+      - Authorize with Satellite client certificate and validate capsule CA from Settings
+    type: bool
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+- name: 'Create Webhook'
+  theforeman.foreman.webhook:
+    username: 'admin'
+    password: 'secret_password'
+    server_url: 'https://foreman.example.com'
+    name: 'test-webhook'
+    target_url: 'https://google.com'
+    http_method: 'GET'
+    event: 'actions.katello.content_view.promote_succeeded'
+    enabled: true
+    organizations:
+      - 'MyOrg'
+    locations:
+      - 'DC1'
+
+- name: 'Remove Webhook'
+  theforeman.foreman.webhook:
+    username: 'admin'
+    password: 'secret_password'
+    server_url: 'https://foreman.example.com'
+    name: 'test-webhook'
+    state: 'absent'
+'''
+
+RETURN = '''
+entity:
+  description: Final state of the affected entities grouped by their type.
+  returned: success
+  type: dict
+  contains:
+    webhooks:
+      description: List of webhooks.
+      type: list
+      elements: dict
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import ForemanTaxonomicEntityAnsibleModule
+
+
+class ForemanWebhookModule(ForemanTaxonomicEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanWebhookModule(
+        foreman_spec=dict(
+            name=dict(required=True),
+            target_url=dict(),
+            http_method=dict(),
```

Comment:
```suggestion
            http_method=dict(choices=['POST', 'GET', 'PUT', 'DELETE', 'PATCH']),
```

(and the same above in the doc)

this will force the user to use a proper http verb :)